### ISSUE TITLE
Course instance copying is hard to discover

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#4638]
Course instances can be copied from their "Settings" tabs, but many users don't discover this and they end up creating a new blank course instance.

We should add a "Copy" button next to each course instance on the "Course Instances" tab.

**Comment by ZacWarham** [comment]
@mwest1066 is this still something that would be of interest to be added?

I [experimented](https://github.com/ZacWarham/PrairieLearn/tree/button-creation) a bit based on the [copy question code](https://github.com/PrairieLearn/PrairieLearn/blob/f2762ababb916f090d70a6d4a98ec9941d7f2e4d/apps/prairielearn/src/components/QuestionContainer.html.ts#L754-L808) and made it a single button instead (as I thought a button for each instance would get messy).

![instances](https://github.com/user-attachments/assets/e40223dc-1268-48f5-ad3a-a29c11d9ff23)

![selection](https://github.com/user-attachments/assets/27d9ed4e-7d3d-40d3-bb1e-7e302008af31)

![final screen](https://github.com/user-attachments/assets/21843eb4-c519-4400-ab48-56aa22442a44)

**Comment by mwest1066** [comment]
I like it! Maybe use the copy icon instead of the plus icon? https://icons.getbootstrap.com/icons/copy/

**Comment by mwest1066** [comment]
I'm also wondering whether we should keep just one "Add course instance" button, but have it open a dialog that has copying as option. Something like this:
```
New course instance
---------------------

Copy all assessments from: [None, start with no assessments] (dropdown listing all other course instances)

Short name for new course instance
[________________]

Long name for new course instance
[________________]

[Cancel] [Make new course instance]
```

The current "copy course instance" button just makes a copy and the relies on the user to rename it. We've been moving away from this UI in favor of dialogs that ask for the name of the new object before creating it. CC @mylesw27 who has been working on this -- maybe Myles already has a new course instance creation dialog? Also CC @nwalters512.

With such a dialog as described above, we could also add a final option in the course instance switcher in @miguelaenlle's new nav like "+ Add new course instance". This could link to the course instances page with the above dialog already open, which I think would help discoverability, because normally I think people don't visit the course instances page very often.

**Comment by ZacWarham** [comment]
Good catch with the copy icon! Happy to see what input others have and a more streamlined creation might be the better approach. It is all practice for me whether it is implemented or not.

One thing I can think of is that making it just copy assessments (for better or worse) would not copy over other settings (access roles, show enrolment and module settings). Perhaps it needs the options for both?

**Comment by mwest1066** [comment]
> One thing I can think of is that making it just copy assessments (for better or worse) would not copy over other settings (access roles, show enrolment and module settings). Perhaps it needs the options for both?

Good point, I guess I meant to copy everything. Probably we should word that just as "Copy existing course instance" or something.

**Comment by nwalters512** [comment]
I sort of like the idea of integrating copying into the "new course instance" flow. That seems like a natural place to put it, and ensures that users don't miss the fact that copying is an option.

I do wonder if there's value in retaining the copy button in some form on the settings page. IMO it's nice to have consistency between different "things" in PL. Questions and assessments are also copyable, and if I'm already looking at a particular question or assessment, I think it's much more natural to try to do the copying from the thing you're already looking at instead of having to navigate back to the question or assessment pages, click "Add ...", and then find the thing you were looking at again in a list of potentially hundreds of things. So, I think we should keep the copy button on the question and assessments settings pages, and if we're keeping it there, then also keep it on the course instance settings page.

I guess what I'm proposing is: have multiple ways to copying something: as part of the "make a new thing" flow, and also from somewhere on the thing you're already looking at.

For lack of a better place to put this comment, we got a report from a user recently that they've gotten confused between copying and editing a few times:

> This [accidentally editing instead of copying] has happened to me by accident on a few occasions at the question level and not on an assessment nor one that is active so I've been very lucky. It seems that the settings page should not be editable and instead have an option for editing or making a copy at the very top. Currently the copy button is at the very bottom and it's so tempting to just jump in and start changing the question or assessment name. Then once you do that you are in big trouble once you make further changes.
>
> _me:_ to clarify, you're suggesting that the settings pages be read-only by default and that you should have to click an "edit" button in order to start making changes?
>
> yes, and right next to that button should be the make a copy button so that it's front and center. if having the edit button isn't a good idea then at least putting the make a copy button up top might make it stand out more.

**Comment by reteps** [comment]
I also brought this up recently. I feel like it should be next to the 'Add course instance' button.
